dds/DCPS/transport/framework/DataLink.inl

```diff
@@ -107,11 +107,14 @@ DataLink::send(TransportQueueElement* element)
 
   element = this->customize_queue_element(element);
   if (!element) {
+    element->data_dropped(true);
```

Comment:
This series of `element->data_dropped(true)` seems to have fixed the memory leak I was looking at. I just wanted to confirm that this is correct and doesn't mess up the element in other places.

Reply:
If element is null (the `if` on 109 succeeds), how can this line invoke a method on it?

Reply:
...I guess it wouldn't be able to. I think the idea I had in mind should have been written like:
```
TransportQueueElement * test_element = this->customize_queue_element(element);
if (test_element)  {
  element = test_element;
} else {
  element->data_dropped(true);
  return;
}
```